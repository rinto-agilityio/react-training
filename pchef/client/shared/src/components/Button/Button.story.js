// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Helper
import { wInfo } from '../../../.storybook/utils';

// Component
import Button from './index'

storiesOf('Button', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <Button
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Success', () => (
    <Button
      type="success"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Danger', () => (
    <Button
      type="danger"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Warning', () => (
    <Button
      type="warning"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Info', () => (
    <Button
      type="info"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))
