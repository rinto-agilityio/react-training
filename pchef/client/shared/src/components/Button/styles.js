// Lib
import { StyleSheet } from 'react-native'

// Themes
import { COLORS, METRICS, FONTS } from '../../themes'

// Create style for button
const styles = StyleSheet.create({
  button: {
    ...METRICS.flexCenter,
    padding: METRICS.padding.sm,
  },
  title: {
    color: COLORS.white,
    textTransform: 'uppercase',
    fontWeight: FONTS.fontWeight.large,
  },
  default: {
    backgroundColor: COLORS.baseBlue,
  },
  success: {
    backgroundColor: COLORS.green,
  },
  danger: {
    backgroundColor: COLORS.red,
  },
  warning: {
    backgroundColor: COLORS.yellow,
  },
  info: {
    backgroundColor: COLORS.darkWhite,
  },
})

export default styles
