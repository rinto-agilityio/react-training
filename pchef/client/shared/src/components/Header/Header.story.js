// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { View } from 'react-native'
import { action } from '@storybook/addon-actions'

// Component
import Header from '.'

// Helper
import { wInfo } from '../../../.storybook/utils';

const props = {
  onPressCategoryIcon: action('clicked category icon'),
  onPressLogo: action('clicked logo'),
}

storiesOf('New feed header', module)
  .addDecorator(wInfo())
  .add('Primary', () => (
    <View style={{ width: 500, margin: 'auto' }}>
      <Header {...props} type="primary" />
    </View>
  ))
  .add('Secondary', () => (
    <Header {...props} onPressCategoryIcon size="secondary" />
  ))
