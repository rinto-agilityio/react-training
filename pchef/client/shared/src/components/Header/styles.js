// Lib
import { StyleSheet } from 'react-native'

// Themes
import { COLORS, METRICS, FONTS } from '../../themes'

// Create style for Header
const styles = StyleSheet.create({
  wrapHeader: {
    backgroundColor: COLORS.black,
  },
  wrapHeaderContent: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  wrapUserInfo: {
    alignSelf: 'center',
    flexDirection: 'row',
  },
  createRecipeBtn: {
    marginRight: METRICS.margin.lg,
  },
  avataDropdown: {
    left: METRICS.position.xl,
    top: METRICS.position.lg,
  },
  container: {
    maxWidth:
      METRICS.screen.width > METRICS.screen.md
        ? METRICS.screen.xl
        : METRICS.screen.width,
    width: '100%',
    marginLeft: 'auto',
    marginRight: 'auto',
  },
  primaryContainer: {
    alignItems: 'flex-start',
    paddingBottom: METRICS.padding.lg,
    paddingTop: METRICS.padding.lg,
    paddingLeft: METRICS.padding.lg + METRICS.padding.md,
  },
  secondaryContainer: {
    paddingLeft: METRICS.padding.lg * 2 + METRICS.padding.md,
    paddingRight: METRICS.padding.lg * 2 + METRICS.padding.md,
    flexDirection: 'row-reverse',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  primaryWrapIcon: {
    marginTop: METRICS.margin.md,
    marginLeft: METRICS.margin.lg,
    marginBottom: METRICS.margin.md,
  },

  text: {
    fontSize: FONTS.fontSize.extraLarge,
    color: COLORS.white,
    fontWeight: FONTS.fontWeight.large,
  },
  image: {
    width: METRICS.image.sm / 2,
    height: METRICS.image.sm / 2,
  },
  secondaryImage: {
    width: METRICS.image.md / 2,
    height: METRICS.image.md / 2,
  },
  secondaryContent: {
    alignItems: 'center',
    flexDirection: 'row-reverse',
    justifyContent: 'space-between',
    paddingRight: METRICS.padding.xl,
  },
  imageWrapper: {
    backgroundColor: COLORS.white,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: METRICS.margin.sm,
    width: METRICS.image.xs,
    height: METRICS.image.xs,
    borderRadius: METRICS.borderRadius.md + METRICS.borderRadius.lg,
  },
  primaryImageWrapper: {
    marginLeft: METRICS.margin.md,
  },
  secondaryImageWrapper: {
    marginTop: METRICS.margin.lg,
    marginBottom: METRICS.margin.lg,
  },
  logo: {
    flexDirection: 'row',
    alignItems: 'center',
  },
})

export default styles
