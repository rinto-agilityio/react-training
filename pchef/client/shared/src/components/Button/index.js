// libs
import React from 'react'
import { Button as ButtonComponent, Text } from 'react-native-paper'

// Styles
import styles from './styles'
import { COLORS } from '../../themes/index'

type Props = {
  onPress?: () => void,
  buttonStyle?: {} | Array<{}>,
  title: string,
  titleStyle?: {} | Array<{}>,
  disabled?: boolean,
  mode?: string,
  contentStyle?: {} | Array<{}>,
  color?: string,
  loading?: boolean,
  type?: string,
}

const Button = ({
  onPress,
  buttonStyle,
  title,
  mode,
  type,
  disabled,
  contentStyle,
  color,
  loading,
  titleStyle,
}: Props) => (
  <ButtonComponent
    color={color}
    loading={loading}
    onPress={onPress}
    mode={mode}
    disabled={disabled}
    style={[styles.button, styles[type], buttonStyle]}
    contentStyle={[styles.content, contentStyle]}
  >
    <Text style={[styles.title, titleStyle]}>{title}</Text>
  </ButtonComponent>
)

Button.defaultProps = {
  onPress: () => {},
  buttonStyle: {},
  titleStyle: {},
  disabled: false,
  mode: 'contained',
  contentStyle: {},
  color: COLORS.baseBlue,
  loading: false,
  type: 'default',
}

export default Button
