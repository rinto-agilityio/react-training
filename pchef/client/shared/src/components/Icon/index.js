// Libs
import React from 'react'
import { View, Text } from 'react-native'
import { IconButton } from 'react-native-paper'

// Styles
import styles from './styles'

// Themes
import { COLORS, FONTS } from '../../themes'

type Props = {
  label?: string | number,
  onPress?: () => void | Promise<void>,
  customStyle?: {} | Array<{}>,
  color?: string,
  size?: number,
  name: string,
  disabled?: boolean,
  wrapperIconStyle?: Object,
  customIconStyles?: {},
};

const Icon = ({
  label,
  onPress,
  customStyle,
  color,
  size,
  name,
  wrapperIconStyle,
  disabled,
  customIconStyles,
}: Props) => (
  <View style={[
    styles.wrapIcon,
    wrapperIconStyle,
  ]}
  >
    <IconButton
      icon={name}
      color={color}
      size={size}
      onPress={onPress}
      disabled={disabled}
      style={[styles.iconStyles, customIconStyles]}
    />
    {label ? <Text style={[styles.label, customStyle]}>{label}</Text> : null}
  </View>
)

Icon.defaultProps = {
  label: '',
  onPress: () => {},
  customStyle: {},
  size: FONTS.fontSize.medium,
  color: COLORS.black,
  wrapperIconStyle: {},
  disabled: false,
  customIconStyles: {},
};

export default Icon
