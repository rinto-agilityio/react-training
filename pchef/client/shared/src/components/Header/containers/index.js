import { compose } from 'react-apollo'

// graphQL query
import { getUser } from '../../../graphql/user'

// component
import Header from '../'

export default compose(getUser)(Header)
