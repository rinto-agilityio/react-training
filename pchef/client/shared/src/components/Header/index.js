// Libs
import React, { useState } from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import { Avatar, Provider, Menu } from 'react-native-paper'

// Common components
import Icon from '../Icon'
import Loading from '../Loading'
import Error from '../Error'
import Button from '../Button'

// Themes
import { COLORS, FONTS } from '../../themes'

// Styles
import styles from './styles'

// Constants
import { URL } from '../../constants/index'

type Props = {
  data: {
    user: {
      avatar: string,
    }
  },
  type?: string,
  onPressLogo?: () => void,
  onPressCategoryIcon?: () => void,
  onDirectTo: (id: string) => void,
}

const Header = ({
  type = 'primary',
  onPressLogo,
  onPressCategoryIcon,
  data = {
    user: {
      avatar: '',
    },
  },
  onDirectTo,
}: Props) => {
  const [visible, setVisible] = useState(false)
  const { user: { avatar } } = data

  return (
    <View style={[styles.wrapHeader, styles[`${type}Container`]]}>
      <View style={styles.container}>
        <Icon
          name="apps"
          size={FONTS.fontSize.extraExtraLarge}
          onPress={onPressCategoryIcon}
          color={COLORS.white}
          wrapperIconStyle={styles.wrapIcon}
        />
        <View style={styles.wrapHeaderContent}>
          <View>
            <TouchableOpacity onPress={onPressLogo} style={styles.logo}>
              <View style={[styles.imageWrapper, styles[`${type}ImageWrapper`]]}>
                <Image
                  source={require('../../assets/images/logo.png')}
                  style={[styles.image, styles[`${type}Image`]]}
                />
              </View>
              <Text style={styles.text}>Cooking</Text>
            </TouchableOpacity>
          </View>
          <View style={styles.wrapUserInfo}>
            <Button
              onPress={() => onDirectTo(URL.CREATE_RECIPE.PATH)}
              title={URL.CREATE_RECIPE.TITLE}
              buttonStyle={styles.createRecipeBtn}
            />
            <Provider>
              <Menu
                style={styles.avataDropdown}
                visible={visible}
                onDismiss={() => setVisible(false)}
                anchor={
                  (
                    <TouchableOpacity onPress={() => setVisible(true)} style={styles.logo}>
                      <Avatar.Image
                        style={styles.pipelineItemImage}
                        size={40}
                        source={{
                          uri: (
                            avatar
                              ? avatar
                              : 'https://www.pngkey.com/png/full/115-1150152_default-profile-picture-avatar-png-green.png'
                          ),
                        }}
                      />
                    </TouchableOpacity>
                  )
                }
              >
                <Menu.Item onPress={() => onDirectTo(URL.PROFILE.PATH)} title={URL.PROFILE.TITLE} />
                <Menu.Item onPress={() => onDirectTo(URL.WISH_LIST.PATH)} title={URL.WISH_LIST.TITLE} />
                <Menu.Item onPress={() => onDirectTo(URL.LOG_OUT.PATH)} title={URL.LOG_OUT.TITLE} />
              </Menu>
            </Provider>
          </View>
        </View>
      </View>
    </View>
  )
}

Header.defaultProps = {
  type: 'primary',
  onPressLogo: () => {},
  onPressCategoryIcon: () => {},
}

export default Header
