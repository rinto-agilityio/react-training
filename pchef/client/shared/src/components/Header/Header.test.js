// Components
import { Menu } from 'react-native-paper'
import Header from '.'
import Icon from '../Icon'
import Button from '../Button'

const props = {
  onPressCategoryIcon: jest.fn(),
  data: {
    user: {
      avatar: 'image.png',
    },
  },
  onDirectTo: jest.fn(),
}

const component = shallow(<Header {...props} />)

describe('Components', () => {
  describe('Header', () => {
    it('Should render Header component', () => {
      const primaryComponent = renderer.create(
        <Header type="primary" customStyle={{ flex: 1 }} />
      )
      const secondaryComponent = renderer.create(<Header type="secondary" />)

      expect(primaryComponent).toMatchSnapshot()
      expect(secondaryComponent).toMatchSnapshot()
    })

    it('Should call to onPressCategoryIcon', () => {
      const icon = component.find(Icon)
      icon.props().onPress()
      expect(props.onPressCategoryIcon).toHaveBeenCalled()
    })

    it('Should call to onDirectTo when press on Button', () => {
      const createRecipeButton = component.find(Button)
      createRecipeButton.props().onPress()
      expect(props.onDirectTo).toHaveBeenCalled()
    })

    it('Should call to onDirectTo when press on Menu.Item', () => {
      const firstMenu = component.find(Menu.Item).at(0)
      firstMenu.props().onPress()
      expect(props.onDirectTo).toHaveBeenCalled()

      const secondMenu = component.find(Menu.Item).at(1)
      secondMenu.props().onPress()
      expect(props.onDirectTo).toHaveBeenCalled()

      const thirdMenu = component.find(Menu.Item).at(2)
      thirdMenu.props().onPress()
      expect(props.onDirectTo).toHaveBeenCalled()
    })

    it('Menu should invisible if trigger onDismiss', () => {
      const menu = component.find(Menu)
      menu.props().onDismiss()
      component.update()

      expect(menu.props().visible).toBe(false)
    })
  })
})
