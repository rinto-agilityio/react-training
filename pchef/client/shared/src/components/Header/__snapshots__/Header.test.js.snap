// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Header Login snapshots 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1habvwh r-backgroundColor-kemksi r-paddingBottom-k8qxaj r-paddingLeft-1yt7n81 r-paddingTop-1knelpx"
>
  <div
    className="css-view-1dbjc4n r-marginLeft-1jj8364 r-marginRight-lchren r-maxWidth-c3zab1 r-width-13qz1uu"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-fontSize-xb2eav"
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="css-hitSlop-mjp8i1"
          style={
            Object {
              "bottom": "-6px",
              "left": "-6px",
              "right": "-6px",
              "top": "-6px",
            }
          }
        />
        <div
          className="css-view-1dbjc4n"
        >
          <div
            className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
            dir="auto"
            focusable={false}
            role="img"
            style={
              Object {
                "WebkitTransform": "scaleX(1)",
                "color": "rgba(255,255,255,1.00)",
                "fontFamily": "MaterialIcons",
                "fontSize": "30px",
                "fontStyle": "normal",
                "fontWeight": "normal",
                "transform": "scaleX(1)",
              }
            }
          >
            î—ƒ
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          tabIndex="0"
        >
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-14lw9ot r-borderRadius-1llaxt6 r-height-uvuy5l r-justifyContent-1777fci r-marginLeft-1n0xq6e r-marginRight-7o8qx1 r-width-cfp7ip"
          >
            <div
              className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
              style={
                Object {
                  "height": "40px",
                  "width": "40px",
                }
              }
            >
              <div
                className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                style={
                  Object {
                    "backgroundImage": "url(\\"logo.png\\")",
                  }
                }
              />
              <img
                alt=""
                className="css-accessibilityImage-9pa8cd"
                draggable={false}
                src="logo.png"
              />
            </div>
          </div>
          <div
            className="css-text-901oao r-color-jwli3a r-fontSize-zz5t5d r-fontWeight-1kfrs79"
            dir="auto"
          >
            Cooking
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n r-alignSelf-1kihuf0 r-flexDirection-18u37iz"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
          style={
            Object {
              "backgroundColor": "rgba(66,165,245,1.00)",
              "borderBottomColor": "rgba(0,0,0,0.00)",
              "borderBottomLeftRadius": "4px",
              "borderBottomRightRadius": "4px",
              "borderBottomStyle": "solid",
              "borderBottomWidth": "0px",
              "borderLeftColor": "rgba(0,0,0,0.00)",
              "borderLeftStyle": "solid",
              "borderLeftWidth": "0px",
              "borderRightColor": "rgba(0,0,0,0.00)",
              "borderRightStyle": "solid",
              "borderRightWidth": "0px",
              "borderTopColor": "rgba(0,0,0,0.00)",
              "borderTopLeftRadius": "4px",
              "borderTopRightRadius": "4px",
              "borderTopStyle": "solid",
              "borderTopWidth": "0px",
              "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
              "marginRight": "20px",
              "minWidth": "64px",
              "paddingBottom": "5px",
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "paddingTop": "5px",
            }
          }
        >
          <div
            className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-overflow-1udh08x r-touchAction-1otgn73 r-userSelect-lrvibr"
            data-focusable={true}
            disabled={false}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              Object {
                "borderBottomLeftRadius": "4px",
                "borderBottomRightRadius": "4px",
                "borderTopLeftRadius": "4px",
                "borderTopRightRadius": "4px",
              }
            }
            tabIndex="0"
          >
            <div
              className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1777fci"
            >
              <div
                className="css-text-901oao css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(0,0,0,1.00)",
                    "direction": "ltr",
                    "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  }
                }
              >
                <span
                  className="css-text-901oao css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
                  dir="auto"
                  style={
                    Object {
                      "direction": "ltr",
                      "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                      "textAlign": "left",
                    }
                  }
                >
                  Create Recipe
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n r-flex-13awgt0"
        >
          <div
            className="css-view-1dbjc4n"
          >
            <div
              className="css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
              data-focusable={true}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              tabIndex="0"
            >
              <div
                className="css-view-1dbjc4n"
                style={
                  Object {
                    "backgroundColor": "rgba(98,0,238,1.00)",
                    "borderBottomLeftRadius": "20px",
                    "borderBottomRightRadius": "20px",
                    "borderTopLeftRadius": "20px",
                    "borderTopRightRadius": "20px",
                    "height": "40px",
                    "width": "40px",
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
                  style={
                    Object {
                      "borderBottomLeftRadius": "20px",
                      "borderBottomRightRadius": "20px",
                      "borderTopLeftRadius": "20px",
                      "borderTopRightRadius": "20px",
                      "height": "40px",
                      "width": "40px",
                    }
                  }
                >
                  <div
                    className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                    style={
                      Object {
                        "backgroundImage": "url(\\"default_avatar.png\\")",
                      }
                    }
                  />
                  <img
                    alt=""
                    className="css-accessibilityImage-9pa8cd"
                    draggable={false}
                    src="default_avatar.png"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components Header Login snapshots 2`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-kemksi r-flexDirection-1euycsn r-justifyContent-1wtj0ep r-paddingLeft-1fd96xs r-paddingRight-yongw"
>
  <div
    className="css-view-1dbjc4n r-marginLeft-1jj8364 r-marginRight-lchren r-maxWidth-c3zab1 r-width-13qz1uu"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-fontSize-xb2eav"
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="css-hitSlop-mjp8i1"
          style={
            Object {
              "bottom": "-6px",
              "left": "-6px",
              "right": "-6px",
              "top": "-6px",
            }
          }
        />
        <div
          className="css-view-1dbjc4n"
        >
          <div
            className="css-text-901oao r-backgroundColor-1niwhzg r-color-jwli3a r-fontSize-xb2eav r-pointerEvents-633pao"
            dir="auto"
            focusable={false}
            role="img"
            style={
              Object {
                "WebkitTransform": "scaleX(1)",
                "fontFamily": "MaterialIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
                "transform": "scaleX(1)",
              }
            }
          >
            î—ƒ
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          tabIndex="0"
        >
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-14lw9ot r-borderRadius-1llaxt6 r-height-uvuy5l r-justifyContent-1777fci r-marginBottom-117bsoe r-marginRight-7o8qx1 r-marginTop-1x0uki6 r-width-cfp7ip"
          >
            <div
              className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
              style={
                Object {
                  "height": "50px",
                  "width": "50px",
                }
              }
            >
              <div
                className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                style={
                  Object {
                    "backgroundImage": "url(\\"logo.png\\")",
                  }
                }
              />
              <img
                alt=""
                className="css-accessibilityImage-9pa8cd"
                draggable={false}
                src="logo.png"
              />
            </div>
          </div>
          <div
            className="css-text-901oao r-color-jwli3a r-fontSize-zz5t5d r-fontWeight-1kfrs79"
            dir="auto"
          >
            Cooking
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n r-alignSelf-1kihuf0 r-flexDirection-18u37iz"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
          style={
            Object {
              "backgroundColor": "rgba(66,165,245,1.00)",
              "borderBottomColor": "rgba(0,0,0,0.00)",
              "borderBottomLeftRadius": "4px",
              "borderBottomRightRadius": "4px",
              "borderBottomStyle": "solid",
              "borderBottomWidth": "0px",
              "borderLeftColor": "rgba(0,0,0,0.00)",
              "borderLeftStyle": "solid",
              "borderLeftWidth": "0px",
              "borderRightColor": "rgba(0,0,0,0.00)",
              "borderRightStyle": "solid",
              "borderRightWidth": "0px",
              "borderTopColor": "rgba(0,0,0,0.00)",
              "borderTopLeftRadius": "4px",
              "borderTopRightRadius": "4px",
              "borderTopStyle": "solid",
              "borderTopWidth": "0px",
              "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
              "marginRight": "20px",
              "minWidth": "64px",
              "paddingBottom": "5px",
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "paddingTop": "5px",
            }
          }
        >
          <div
            className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-overflow-1udh08x r-touchAction-1otgn73 r-userSelect-lrvibr"
            data-focusable={true}
            disabled={false}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              Object {
                "borderBottomLeftRadius": "4px",
                "borderBottomRightRadius": "4px",
                "borderTopLeftRadius": "4px",
                "borderTopRightRadius": "4px",
              }
            }
            tabIndex="0"
          >
            <div
              className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1777fci"
            >
              <div
                className="css-text-901oao css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(0,0,0,1.00)",
                    "direction": "ltr",
                    "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  }
                }
              >
                <span
                  className="css-text-901oao css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
                  dir="auto"
                  style={
                    Object {
                      "direction": "ltr",
                      "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                      "textAlign": "left",
                    }
                  }
                >
                  Create Recipe
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n r-flex-13awgt0"
        >
          <div
            className="css-view-1dbjc4n"
          >
            <div
              className="css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
              data-focusable={true}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              tabIndex="0"
            >
              <div
                className="css-view-1dbjc4n"
                style={
                  Object {
                    "backgroundColor": "rgba(98,0,238,1.00)",
                    "borderBottomLeftRadius": "20px",
                    "borderBottomRightRadius": "20px",
                    "borderTopLeftRadius": "20px",
                    "borderTopRightRadius": "20px",
                    "height": "40px",
                    "width": "40px",
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
                  style={
                    Object {
                      "borderBottomLeftRadius": "20px",
                      "borderBottomRightRadius": "20px",
                      "borderTopLeftRadius": "20px",
                      "borderTopRightRadius": "20px",
                      "height": "40px",
                      "width": "40px",
                    }
                  }
                >
                  <div
                    className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                    style={
                      Object {
                        "backgroundImage": "url(\\"default_avatar.png\\")",
                      }
                    }
                  />
                  <img
                    alt=""
                    className="css-accessibilityImage-9pa8cd"
                    draggable={false}
                    src="default_avatar.png"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
